File: src/main/java/net/dontdrinkandroot/gitki/wicket/model/user/UserLastNameModel.java
package net.dontdrinkandroot.gitki.wicket.model.user;

import net.dontdrinkandroot.gitki.model.User;
import net.dontdrinkandroot.wicket.model.AbstractChainedModel;
import org.apache.wicket.model.IModel;

/**
 * @author Philip Washington Sorst <philip@sorst.net>
 */
public class UserLastNameModel extends AbstractChainedModel<User, String>
{
    public UserLastNameModel(IModel<? extends User> parent)
    {
        super(parent);
    }

    @Override
    public String getObject()
    {
        return this.getParentObject().getLastName();
    }

    @Override
    public void setObject(String object)
    {
        this.getParentObject().setLastName(object);
    }
}

File: src/main/kotlin/net/dontdrinkandroot/gitki/wicket/model/user/UserLastNameModel.kt
package net.dontdrinkandroot.gitki.wicket.model.user

import net.dontdrinkandroot.gitki.model.User
import net.dontdrinkandroot.wicket.model.AbstractChainedModel
import org.apache.wicket.model.IModel

class UserLastNameModel(parent: IModel<out User>) : AbstractChainedModel<User, String>(parent) {

    override fun getObject(): String {
        return this.parentObject!!.lastName!!
    }

    override fun setObject(`object`: String) {
        this.parentObject!!.lastName = `object`
    }
}